Framework/PythonInterface/plugins/algorithms/MedianBinWidth.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.api import AlgorithmFactory, HistogramValidator,\
+    MatrixWorkspaceProperty, PythonAlgorithm
+from mantid.kernel import Direction
+import numpy
+import roundinghelper
+
+
+class MedianBinWidth(PythonAlgorithm):
+
+    _PROP_BIN_WIDTH = 'BinWidth'
+    _PROP_INPUT_WS = 'InputWorkspace'
+
+    def category(self):
+        '''
+        Return algorithm's category.
+        '''
+        return 'Utility\\Calculation'
+
+    def name(self):
+        '''
+        Return algorithm's name.
+        '''
+        return 'MedianBinWidth'
+
+    def summary(self):
+        '''
+        Return algorithm's summary.
+        '''
+        return ("Calculates the average of workspace's histograms'"
+                " median bin widths.")
+
+    def version(self):
+        '''
+        Return algorithm's version.
+        '''
+        return 1
+
+    def PyInit(self):
+        '''
+        Declares algorithm's properties.
+        '''
+        self.declareProperty(
+            MatrixWorkspaceProperty(name=self._PROP_INPUT_WS,
+                                    defaultValue='',
+                                    validator=HistogramValidator(),
+                                    direction=Direction.Input),
+            doc='The workspace containing the input data')
+        roundinghelper.declare_rounding_property(self)
+        self.declareProperty(self._PROP_BIN_WIDTH,
+                             defaultValue=0.0,
+                             direction=Direction.Output,
+                             doc='The averaged median bin width')
+
+    def PyExec(self):
+        '''
+        Averages the median bin widths of the input workspace.
+        '''
+        inputWs = self.getProperty(self._PROP_INPUT_WS).value
+        roundingMode = self.getProperty(
+            roundinghelper.PROP_NAME_ROUNDING_MODE).value
+        n = inputWs.getNumberHistograms()
+        medians = numpy.empty(n)
+        for wsIndex in range(n):
```

Comment:
```
from mantid.simpleapi import *
import numpy as np

x=Load(Filename='/SNS/HYS/IPTS-16818/shared/autoreduce/4pixel/HYS_149202_4pixel.nxspe')
def loopway(ws):
    n=ws.getNumberHistograms()
    medians=np.empty(n)
    for wsIndex in range(n):
        xs=ws.readX(wsIndex)
        dxs=xs[1:]-xs[:-1]
        medians[wsIndex]=np.median(dxs)
    return medians

def npway(ws):
    x2D=ws.extractX()
    dx2D=np.diff(x2D)
    medians=np.median(dx2D,axis=1)
    return medians
```
I did some timing tests. The NXSPE file is 5120 spectra, 442 bins
``` 
%timeit loopway(x)
1 loop, best of 3: 317 ms per loop
%timeit npway(x)
10 loops, best of 3: 35.9 ms per loop
```
Almost a factor of 10 increase in speed